cmd/qri.go

```diff
@@ -220,93 +152,106 @@ func (o *QriOptions) CryptoGenerator() gen.CryptoGenerator {
 
 // RPC returns from internal state
 func (o *QriOptions) RPC() *rpc.Client {
-	return o.rpc
+	if err := o.Init(); err != nil {
+		return nil
+	}
+	return o.inst.RPC()
 }
 
 // ConnectionNode returns the internal QriNode, if it is available
 func (o *QriOptions) ConnectionNode() (*p2p.QriNode, error) {
-	if o.repo == nil {
+	if o.inst == nil {
 		return nil, fmt.Errorf("repo not available")
 	}
-	return o.node, nil
+	return o.inst.Node(), nil
 }
 
 // DatasetRequests generates a lib.DatasetRequests from internal state
 func (o *QriOptions) DatasetRequests() (*lib.DatasetRequests, error) {
 	if err := o.Init(); err != nil {
 		return nil, err
 	}
-	return lib.NewDatasetRequests(o.node, o.rpc), nil
+	return lib.NewDatasetRequests(o.inst.Node(), o.inst.RPC()), nil
```

Comment:
Please add a TODO about switches these implementations to passing in an `instance` without pulling it apart.

Reply:
sure thing, I added TODOs at all struct definition sites in the lib package